pyresample/resampler.py

```diff
@@ -132,3 +148,193 @@ def _create_cache_filename(self, cache_dir=None, prefix='',
+        name = f"{funcname(funk)}-{token}"
+    return name
+
+
+def _make_fill_value(fill_value, dtype):
+    if fill_value is None:
+        if np.issubdtype(dtype, np.integer):
+            fill_value = np.iinfo(dtype).min
+        else:
+            fill_value = np.nan
+    return fill_value
+
+
+def _check_resolution_mismatch(src_area_crop, dtype):
+    res_chunks, _ = _normalize_chunks_for_area(src_area_crop, dask.config.get('array.chunk-size', '128MiB'),
+                                               dtype)
+    if len(res_chunks[0]) * len(res_chunks[1]) >= 4:
+        logger.warning("The input area chunks are large. "
+                       "This usually means that the input area is of much higher resolution than the output "
+                       "area. You can reduce the chunks passed, and ponder whether you are using the right "
+                       "resampler for the job.")
+
+
+def _create_task(funk, smaller_src_arrays, src_block_info, dst_arrays, dst_block_info, position, fill_value,
+                 dependencies, kwargs):
+    """Create a task for resample_blocks."""
+    from dask.utils import has_keyword
+    args = []
+    for smaller_data in smaller_src_arrays:
+        args.append((smaller_data.name, *([0] * smaller_data.ndim)))
+        dependencies.append(smaller_data)
+    for dst_array in dst_arrays:
+        dst_position = [0] * (dst_array.ndim - 2) + list(position[-2:])
+        args.append((dst_array.name, *dst_position))
+    funk_kwargs = kwargs.copy()
+    funk_kwargs['fill_value'] = fill_value
+    if has_keyword(funk, "block_info"):
+        funk_kwargs["block_info"] = {0: src_block_info,
+                                     None: dst_block_info}
+    pfunk = partial(funk, **funk_kwargs)
+    task = (pfunk, *args)
+    return task
+
+
+def crop_data_around_area(source_geo_def, src_arrays, target_geo_def):
+    """Crop the data around the provided area."""
+    small_source_geo_def, x_slice, y_slice = crop_source_area(source_geo_def, target_geo_def)
+    smaller_src_arrays = []
+    for data in src_arrays:
+        smaller_src_arrays.append(data[..., y_slice, x_slice].rechunk([-1] * data.ndim))
+
+    block_info = {"shape": source_geo_def.shape,
+                  "array-location": (y_slice, x_slice),
+                  "area": small_source_geo_def}
+    return smaller_src_arrays, small_source_geo_def, block_info
+
+
+@lru_cache
+def crop_source_area(source_geo_def, target_geo_def):
+    """Crop a source area around the provided target area."""
+    slicer = create_slicer(source_geo_def, target_geo_def)
+    x_slice, y_slice = slicer.get_slices()
+    small_source_geo_def = source_geo_def[y_slice, x_slice]
+    if isinstance(small_source_geo_def, SwathDefinition):
+        small_source_geo_def.lons.data = small_source_geo_def.lons.data.rechunk((-1, -1))
+        small_source_geo_def.lats.data = small_source_geo_def.lats.data.rechunk((-1, -1))
+    return small_source_geo_def, x_slice, y_slice
```

Comment:
What part of this function takes a long time that caching is necessary? Additionally, I'm a little nervous about the inplace assignment to `.data` in the SwathDefinition case...oh nevermind, that area is the cropped version already.